### DISCUSSION TITLE
How to create a Mock object of ipcMain to run tests

### DISCUSSION THREAD

**Discussion by Ghost295** [cawa-93/vite-electron-builder#726] [Q&A] [answered]
IpcMain is undefined when the tests run mainWindow.ts.
I could not find a way to create a mock object of ipcMain to extend vi.fn().

Using `const ipcMain = vi.fn() as MaybeMocked<typeof ipcMain.on>;` or `vi.fn().mockImplementation(ipcMain.on)`

What am I missing?

**Comment by cawa-93** [comment, marked as answer]
You should mock all `electron` module exports. Example for `BrowserWindow` class: https://github.com/cawa-93/vite-electron-builder/blob/b5b7282edb9d7286c0ed5dc97703d90e128eba48/packages/main/tests/unit.spec.ts#L10-L24
